syntax = "proto2";
option optimize_for = LITE_RUNTIME;

message Foo {
    optional fixed64 id = 1;
    optional int32 count = 2;
    optional int32 prefix = 3;
    optional int32 length = 4;
}

message Bar {
    optional Foo parent = 1;
    optional int32 time = 2;
    optional float ratio = 3;
    optional int32 size = 4;
}

message FooBar {
    optional Bar sibling = 1;
    optional string name = 2;
    optional double rating = 3;
    optional int32 postfix = 4;
}

enum Fruit {
    FRUIT_APPLES = 1;
    FRUIT_PEARS = 2;
    FRUIT_BANANAS = 3;
}

message FooBarContainer {
    repeated FooBar list = 1;
    optional bool initialized = 2;
    optional Fruit fruit = 3;
    optional string location = 4;
}